### PR TITLE
feat(privatek8s): add Jenkins controller

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR add the Jenkins controller intended for infra.ci.jenkins.io, without the corresponding jobs definitions (for another PR)

Follow-up of https://github.com/jenkins-infra/kubernetes-management/pull/3247
Remaining changes of https://github.com/jenkins-infra/kubernetes-management/pull/3194
Ref: https://github.com/jenkins-infra/helpdesk/issues/2844

### CODE REVIEW

**Comment by lemeurherve** [comment]
LGTM

image

File: clusters/privatek8s.yaml [hunk 1]
```diff
@@ -43,6 +46,34 @@ releases:
       - "../config/ext_datadog_privatek8s.yaml"
     secrets:
       - "../secrets/config/datadog/privatek8s-secrets.yaml"
+  - name: jenkins-infra
+    namespace: jenkins-infra
+    chart: jenkins/jenkins
+    version: 4.2.17
+    needs:
+      - jenkins-infra-additional # Required to allow the Jenkins controller to spawn kubernetes pods in the current cluster
```
Comment (dduportal) [inline clusters/privatek8s.yaml:54]: I'm not sure to understand the meaning of this comment. Is this a copy and paste from the current infra.ci (temp-privatek8s) manifest?

**Review by dduportal — APPROVED** [review]
Please note that this PR, once merged, could break infra.ci (due to the credential name change to connect to Kubernetes).

- Please advertise in IRC #jenkins-infra at least 10 min before merging so other users will know that infra.ci will go down
- No worries if it does not work once back online: you'll have to, eventually, apply credentials manually (or use service account in the pod)
If this is the case, no

File: clusters/privatek8s.yaml (hunk 1 above)
Comment (lemeurherve) [reply clusters/privatek8s.yaml:54]: It's replicated from the current infra.ci: https://github.com/jenkins-infra/kubernetes-management/blob/main/clusters/temp-privatek8s.yaml#L53-L54

**Pull request merged**